core/src/com/unciv/ui/objectdescriptions/BuildingDescriptions.kt

```diff
@@ -93,12 +80,38 @@ object BuildingDescriptions {
-            translatedLines += "\n" +
-                "[${city.civ.getEquivalentBuilding(missingUnique!!.params[0])}] required:".tr() +
-                " " + missingCities.joinToString(", ") { it.name.tr(hideIcons = true) }
-            // Can't nest square bracket placeholders inside curlies, and don't see any way to define wildcard placeholders. So run translation explicitly on base text.
+        if (showAdditionalInfo) {
+            // Inefficient in theory. In practice, buildings seem to have only a small handful of uniques.
+            for (unique in building.uniqueObjects) {
+                // TODO: Unify with rejection reasons?
+                if (unique.type == UniqueType.RequiresBuildingInAllCities) {
+                    val missingCities = city.civ.cities.filterNot {
+                        it.isPuppet
+                            || it.cityConstructions.containsBuildingOrEquivalent(unique.params[0])
+                    }
+                    // Could be red. But IMO that should be done by enabling GDX's ColorMarkupLanguage globally instead of adding a separate label.
+                    if (missingCities.isNotEmpty()) translatedLines += "\n" +
+                        "[${city.civ.getEquivalentBuilding(unique.params[0])}] required:".tr() +
+                        " " + missingCities.joinToString(", ") { it.name.tr(hideIcons = true) }
+                    // Can't nest square bracket placeholders inside curlies, and don't see any way to define wildcard placeholders. So run translation explicitly on base text.
+                }
+
+                else if (unique.type == UniqueType.OnlyAvailable)
+                    for (conditional in unique.conditionals) {
+                        // TODO: Unify with rejection reasons?
+                        if (conditional.type == UniqueType.ConditionalBuildingBuiltAll) {
+                            val missingCities = city.civ.cities.filterNot {
+                                it.cityConstructions.containsBuildingOrEquivalent(conditional.params[0])
+                                    || !it.matchesFilter(conditional.params[1])
+                            }
+                            // Could be red. But IMO that should be done by enabling GDX's ColorMarkupLanguage globally instead of adding a separate label.
+                            if (missingCities.isNotEmpty()) translatedLines += "\n" +
+                                "[${city.civ.getEquivalentBuilding(unique.params[0])}] required:".tr() +
+                                " " + missingCities.joinToString(", ") { it.name.tr(hideIcons = true) }
+                            // Can't nest square bracket placeholders inside curlies, and don't see any way to define wildcard placeholders. So run translation explicitly on base text.
+                        }
+                    }
+            }
```

Comment:
This really feels like a separate function

Reply:
Not sure what the fix is for the "text should be red" comment. Tbh, put basically no effort into looking for the solution, though. `FormattedLine`? Nah, needs to be a string. Oh... huh, actually, this just doesn't support color because of that fact. Interesting

Reply:
Future me's problem. We're still missing a few stuff I the description of differences for replacement units such as tech differences, great person points, or specialists. I'm willing to call that as not relevant to this pr but something to note for later